tracee/tracee.go

```diff
@@ -75,28 +74,15 @@ func (tc TraceeConfig) Validate() error {
 	return nil
 }
 
-// This var is supposed to be injected *at build time* with the contents of the ebpf c program
-var ebpfProgramBase64Injected string
-
-func getEBPFProgram() (string, error) {
-	// if there's a local file, use it
+func getEBPFObjectPath() (string, error) {
 	exePath, err := os.Executable()
 	if err != nil {
 		return "", err
 	}
-	ebpfFilePath := filepath.Join(filepath.Dir(exePath), "./event_monitor_ebpf.c")
+	ebpfFilePath := filepath.Join(filepath.Dir(exePath), "./event_monitor_ebpf.o")
 	_, err = os.Stat(ebpfFilePath)
 	if !os.IsNotExist(err) {
-		p, err := ioutil.ReadFile(ebpfFilePath)
-		return string(p), err
-	}
-	// if there's no local file, try injected variable
-	if ebpfProgramBase64Injected != "" {
-		p, err := base64.StdEncoding.DecodeString(ebpfProgramBase64Injected)
```

Comment:
I think we should keep this logic for the reasons I described in the makefile (calling clang on the target host)